test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Client/ProxyExecutionManagerTests.cs

```diff
@@ -338,6 +340,22 @@ public void SetupChannelShouldThrowExceptionIfClientConnectionTimeout()
             Assert.ThrowsException<TestPlatformException>(() => this.testExecutionManager.SetupChannel(new List<string> { "source.dll" }));
         }
 
+        [TestMethod]
+        public void SetupChannelShouldThrowExceptionIfTestHostExitedBeforeConnectionIsEstablished()
+        {
+            this.mockRequestSender.Setup(s => s.WaitForRequestHandlerConnection(It.IsAny<int>(), It.IsAny<CancellationToken>())).Returns(false);
+            this.mockTestHostManager.Setup(tmh => tmh.LaunchTestHostAsync(It.IsAny<TestProcessStartInfo>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(true)).Callback(() => { this.mockTestHostManager.Raise(t => t.HostExited += null, new HostProviderEventArgs("I crashed!")); });
+
+            try
+            {
+                this.testExecutionManager.SetupChannel(new List<string> { "source.dll" });
+            }
+            catch (TestPlatformException tpe)
+            {
+                Assert.AreEqual(string.Format(CrossPlatEngineResources.Resources.TestHostExitedWithError, "I crashed!"), tpe.Message);
+            }
```

Comment:
Use Assert.ThrowException, it returns message, assert for message separately if you want.

Reply:
Didn't know about the return value. Thanks. will make this change